### PR TITLE
Changed request type to GraphQLMode

### LINKED ISSUE #5: Research about PokeAPI and its filters
Make a research of how to obtain the filters from the PokeAPI, which library to use and which methods will need to be required for filtering pokemon.
### Acceptance Criteria

- [x] Find a way to apply filters on the pokeapi.
- [x] Document that finding on the Wiki page.
- [x] Show an example on the class

### Epic Link
[Pokemon List Epic](https://github.com/jmacboy/Pokedex-ios/wiki/Epic-for-Pokemon-list)

### PR DESCRIPTION
**Pull request by YawarValeriano**
Changed files in service structure
Added fiedl in PokemonRaw to set image URL
Closes #5

### CODE REVIEW

File: Pokedex_ios/Helpers/GraphQLHelpers/GraphQLResultConverter.swift [hunk 1]
```diff
@@ -0,0 +1,28 @@
+//
+//  GraphQLResultConverter.swift
+//  Pokedex_ios
+//
+//  Created by admin on 7/1/22.
```
Comment (jmacboy) [inline Pokedex_ios/Helpers/GraphQLHelpers/GraphQLResultConverter.swift]: Please add your name to the templates if possible

File: Pokedex_ios/Helpers/GraphQLHelpers/GraphQLResultConverter.swift [hunk 2]
```diff
@@ -0,0 +1,28 @@
...
+class ConversionHelper {
+    static let shared = ConversionHelper()
+    
+    func convertFromDataToPokemonStruct(data: HomePageQuery.Data, completion: @escaping(Result<[PokemonRaw], Error>) -> Void) {
+        do {
+            let jsonData = try JSONSerialization.data(withJSONObject: data.jsonObject)
+            var decode: DataClass = try JSONDecoder().decode(DataClass.self, from: jsonData)
+            for index in 0..<decode.species.count {
+        
+                decode.species[index].urlImage = "https://raw.githubusercontent.com/PokeAPI/sprites/master/sprites/pokemon/other/official-artwork/\(decode.species[index].id).png"
```
Comment (jmacboy) [inline Pokedex_ios/Helpers/GraphQLHelpers/GraphQLResultConverter.swift]: Use this url as a constant variable

File: Pokedex_ios/Helpers/GraphQLHelpers/queries.graphql [hunk 3]
```diff
@@ -0,0 +1,21 @@
+query homePage{
```
Comment (jmacboy) [inline Pokedex_ios/Helpers/GraphQLHelpers/queries.graphql]: this should be in another package, in helpers doesn´t sound right, maybe data or services.

**Review by jmacboy — CHANGES_REQUESTED** [review]

File: Pokedex_ios/Helpers/NetworkManager.swift [hunk 4]
```diff
@@ -10,7 +10,9 @@ import Apollo
 
 class NetworkManager {
     
+    private let pokeApiEndpoint = "https://beta.pokeapi.co/graphql/v1beta"
```
Comment (jmacboy) [inline Pokedex_ios/Helpers/NetworkManager.swift]: Move this to a global constants file (discuss with your coworkers where will be the best package for that as you three have the same issues

**Review by jmacboy — CHANGES_REQUESTED** [review]

File: Pokedex_ios/Helpers/GraphQLHelpers/GraphQLResultConverter.swift (hunk 2 above)
Comment (YawarValeriano) [reply Pokedex_ios/Helpers/GraphQLHelpers/GraphQLResultConverter.swift]: Fixed in last commit

File: Pokedex_ios/Helpers/NetworkManager.swift (hunk 4 above)
Comment (YawarValeriano) [reply Pokedex_ios/Helpers/NetworkManager.swift]: Fixed in last commit

**Comment by jmacboy** [comment]
@YawarValeriano please pull the latest changes from develop

File: Pokedex_ios/Services/PokedexService.swift [hunk 5]
```diff
@@ -0,0 +1,41 @@
...
+
+class PokedexService: PokedexServiceProtocol {
+    func getPokemons(completion: @escaping (Result<[PokemonRaw], Error>) -> Void) {
+        
+        NetworkManager.shared.apollo.fetch(query: HomePageQuery()) { data in
+            switch data {
+            case .success(let graphResult):
+                ConversionHelper.shared.convertFromDataToPokemonStruct(data: graphResult.data!) { data in
+                    switch data {
+                    case.success(let species):
```
[the remark this answers is not in the document]
Comment (YawarValeriano) [reply Pokedex_ios/Services/PokedexService.swift:23]: graphResult.data is being converted to an array of species using the ConversionHelper. Maybe a best name would be "convertedPokemonSpecies"

**Pull request merged**